src/Widgets/TerminalWidget.vala

```diff
@@ -103,6 +102,8 @@ namespace PantheonTerminal {
             private set;
         }
 
+        public double zoom_factor;
```

Comment:
This needs to be a property and the setter clause needs to contain code that will change the size of the font when the factor changes (with proper checks).  The increment, decrement and set_default zoom functions need to be modified accordingly (they only need change the zoom_factor).